### PR TITLE
KAFKA-4458 add per partition in-sync and assigned replica count

### PR DESCRIPTION
**Pull request by xvrl**

### CODE REVIEW

File: core/src/main/scala/kafka/cluster/Partition.scala [hunk 1]
```diff
@@ -74,6 +74,34 @@ class Partition(val topic: String,
...
+                 case Some(_) =>
+                   assignedReplicas.size
+                 case None =>
+                   0
+               }
+             }
+           },
+           tags
+  )
+
```
Comment (ijuma) [inline core/src/main/scala/kafka/cluster/Partition.scala:94]: The indenting doesn't seem right. We use 2 spaces for Scala.

Comment (ijuma) [reply core/src/main/scala/kafka/cluster/Partition.scala:94]: I think I'd do it like:

```scala
newGauge("InSyncReplicasCount",
    new Gauge[Int] {
      def value = {
        if (isLeaderReplicaLocal) inSyncReplicas.size
        else 0
      }
    },
    tags
  )

  newGauge("ReplicasCount",
    new Gauge[Int] {
      def value = {
        if (isLeaderReplicaLocal) assignedReplicas.size
        else 0
      }
    },
    tags
  )


  newGauge("UnderReplicated",
    new Gauge[Int] {
      def value = {
        if (isUnderReplicated) 1 else 0
      }
    },
    tags
  )

  def isUnderReplicated: Boolean =
    isLeaderReplicaLocal && inSyncReplicas.size < assignedReplicas.size

  private def isLeaderReplicaLocal: Boolean = leaderReplicaIfLocal.isDefined
```

It's a bit more concise.

Comment (xvrl) [reply core/src/main/scala/kafka/cluster/Partition.scala:94]: Happy to make the change. I was just mimicking the existing code to be consistent.

Comment (ijuma) [reply core/src/main/scala/kafka/cluster/Partition.scala:94]: Thanks. I actually have a branch with clean-ups to this class as it's unnecessarily verbose in a number of places.

**Comment by ijuma** [comment]
Thanks for the PR, LGTM.

**Pull request closed without merging**